sql-plugin/src/main/scala/com/nvidia/spark/rapids/ShuffleBufferCatalog.scala

```diff
@@ -201,6 +201,15 @@ class ShuffleBufferCatalog(
     val shuffleBufferId = getShuffleBufferId(tableId)
     acquireBuffer(shuffleBufferId)
   }
+
+  /**
+    * Remove a buffer and table given a buffer ID
+    * @param id buffer identifier
+    */
+  def removeBuffer(id: ShuffleBufferId): Unit = {
+    tableMap.remove(id.tableId)
+    catalog.removeBuffer(id)
```

Comment:
Without any synchronization while modifying two related concurrent maps, is the assumption nobody should be asking for this shuffle buffer ID at the point this is called?  Elsewhere in the code that does this there's a comment explaining why there's no synchronization, wondering if we either have a sync issue here or should have a similar comment explaining why sync is unnecessary.

Reply:
So yes, the assumption is that the buffer is not being asked by anyone else, since it hasn't been advertised to the tracker of its existence. That said, it could be getting spilled at the same time, so I think there could be a race there.

The iterator has a very similar call. Where it will remove a received buffer from the `ShuffleReceivedBufferCatalog` when handing it off to the task (and the implementation is the same).

I am not sure at this stage is spilling is an issue here, do you have concerns with that case? I'll look more at that path too.

Reply:
I don't think spilling is an issue here.  `RapidsBufferCatalog` should already properly handle race conditions where a buffer mapping is being updated via spill just as it is being removed separately from what `ShuffleReceivedBufferCatalog` does.  A spill doesn't change shuffle block IDs or table IDs so `ShuffleReceivedBufferCatalog` doesn't know or care that spills are occuring -- that's an implementation detail hidden by `RapidsBufferCatalog`.

So if the synchronization isn't necessary because we fully expect no more than one thread to ever request a particular buffer, we can cover that with a comment here and/or at the top of the shuffle buffer catalog class.